tests/test_basic_logging.py

```diff
@@ -0,0 +1,112 @@
+    assert bch.logger.getEffectiveLevel() == logging.INFO
+
+    # application specific settings for fenicsxconcrete
+    set_log_levels({bch.logger.name: logging.DEBUG})
+    bch.logger.debug("Hello, Debugger!")
+    assert bch.logger.getEffectiveLevel() == logging.DEBUG
+
+
+def test_fenicsx_loggers():
+    """application specific settings for FEniCSx"""
+
+    # ### ufl and ffcx
+    ufl_logger = ufl.log.get_logger()
+    # it seems per default the levels are
+    # ufl: DEBUG (10)
+    # ffcx: WARNIG (30)
+    assert ffcx.logger.getEffectiveLevel() == logging.WARNING
+    assert ufl_logger.getEffectiveLevel() == logging.DEBUG
+
+    set_log_levels(
+        {ffcx.logger.name: logging.ERROR, ufl_logger.name: logging.CRITICAL}
+            )
+
+    assert ffcx.logger.getEffectiveLevel() == logging.ERROR
+    assert ufl_logger.getEffectiveLevel() == logging.CRITICAL
+
+    # ### dolfinx
+    initial_level = dolfinx.log.get_log_level()
+    assert initial_level.value == -1  # WARNING
+
+    # dolfinx.log.set_log_level() only accepts dolfinx.log.LogLevel
+    with pytest.raises(TypeError):
+        dolfinx.log.set_log_level(-1)
+    with pytest.raises(TypeError):
+        dolfinx.log.set_log_level(logging.INFO)
+    with pytest.raises(TypeError):
+        dolfinx.log.set_log_level("INFO")
+
+    dolfinx.log.set_log_level(dolfinx.log.LogLevel.ERROR)
+    current_level = dolfinx.log.get_log_level()
+    assert current_level.value == -2
+
+    # note that dolfinx.log.LogLevel has levels INFO, WARNING, ERROR and OFF
+    # and that the integer values do not follow the convention of the stdlib
+    # logging
+    dfx_levels = [
+        (dolfinx.log.LogLevel.INFO, 0),
+        (dolfinx.log.LogLevel.WARNING, -1),
+        (dolfinx.log.LogLevel.ERROR, -2),
+        (dolfinx.log.LogLevel.OFF, -9),
+    ]
+    for lvl, value in dfx_levels:
+        dolfinx.log.set_log_level(lvl)
+        assert dolfinx.log.get_log_level().value == value
+
+
+@pytest.mark.parametrize("setup", [CantileverBeam,
+                                   TensileBeam,
+                                   CompressionCylinder])
+def test_fenicsxconcrete_example(setup):
+    """Shows how to set log levels for fenicsxconcrete example"""
+
+    setup_parameters = setup.default_parameters()
+
+    # initialize with default parameters
+    experiment = setup(setup_parameters)
+    default_param = LinearElasticity.default_parameters()[1]
+    problem = LinearElasticity(experiment, default_param)
+
+    for thing in [problem, experiment]:
+        assert thing.logger.getEffectiveLevel() == logging.INFO
+
+    # set log level of each class instance
+    # using `set_log_levels`
+    my_levels = {problem.logger.name: logging.ERROR,
+                 experiment.logger.name: logging.DEBUG}
+    set_log_levels(my_levels)
+
+    assert problem.logger.getEffectiveLevel() == logging.ERROR
+    assert experiment.logger.getEffectiveLevel() == logging.DEBUG
```

Comment:
maybe this is my lack of knowledge or me being unfamiliar with how logging works, but why do we now set different things for the problem and the experiment?

Reply:
I guess we are talking about different use cases for this feature and maybe what I have in mind is something different.
I had the person in mind who might use one of our simulations without really knowing what is going on, not some involved developer who want to do detailed debugging.
The log_level would be a possibility to control the output of our package.

Reply:
Maybe it would be possible to add a possible (standard) logger set-up in the example, which is planned for the documentation.

Reply:
> I had the person in mind who might use one of our simulations without really knowing what is going on

in this case the default should be good?
But, I agree that it should be possible to set the level once for the entire package (all modules).
The use case I pointed out should also be possible and would make our life easier I guess.

Reply:
> Maybe it would be possible to add a possible (standard) logger set-up in the example, which is planned for the documentation.

yes, this is a better form to document the usage than this test.
maybe, we can do this once we have agreed on the basic structure?